tests/lib/rules/no-restricted-imports.js

```diff
@@ -430,6 +430,38 @@ ruleTester.run("no-restricted-imports", rule, {
                     allowImportNamePattern: "^Foo"
                 }]
             }]
+        },
+        {
+            code: "import withPatterns from \"foo/bar\";",
+            options: [{ patterns: [{ regex: "foo/(?!bar)", message: "foo is forbidden, use bar instead" }] }]
+        },
+        {
+            code: "import withPatternsCaseSensitive from 'foo';",
+            options: [{
+                patterns: [{
+                    regex: "FOO",
+                    message: "foo is forbidden, use bar instead",
+                    caseSensitive: true
+                }]
+            }]
+        },
```

Comment:
worth adding a test `caseSensitive: false` to test the default?

Reply:
Added in https://github.com/eslint/eslint/pull/18622/commits/8f73b4a0a18672f7b521ac8b95d27c1f14928987